aws/tf/modules/sra/network.tf

```diff
@@ -67,4 +70,15 @@ resource "aws_security_group" "sg" {
   tags = {
     Name = "${var.resource_prefix}-data-plane-sg"
   }
+}
+
+resource "aws_security_group_rule" "esc_conditional_egress" {
```

Comment:
Looks like [AWS advises against](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/security_group_rule) using a `aws_security_group_rule` with an `aws_security_group` that contains inline rules like the `sg` resource above.  One way to work around the issue is to remove the `aws_security_group_rule` and add another `dynamic` block to `sg` where the `for_each` is something like this:  `for_each = [for x in [var.compliance_security_profile ? 2443 : null]: x]`.  It isn't the most aesthetically pleasing since TF isn't great with conditional logic.

Reply:
good call - let me explore this